src/fragments/lib/auth/common/mfa/flows.mdx

```diff
@@ -0,0 +1,214 @@
+
+**`setUpTOTP` API**
+
+import iosSetUpTOTP from "/src/fragments/lib/auth/ios/totp/set_up_totp.mdx";
+
+<Fragments fragments={{ios: iosSetUpTOTP}} />
+
+Once the Authenticator app is set up, the user must generate a TOTP code and provide it to the library. Pass the code to `verifyTOTPSetup` to complete the TOTP setup process.
+
+**`verifyTOTPSetup` API**
+
+import iosVerifyTOTPSetup from "/src/fragments/lib/auth/ios/totp/verify_totp_setup.mdx";
+
+<Fragments fragments={{ios: iosVerifyTOTPSetup}} />
+
+### Recovering from a lost TOTP device
+
+<Callout warning>
+
+If a user loses access to their TOTP device, they would need to contact an administrator to help get access to their account. Based on the Cognito User Pool configuration, the administrator can use the [AdminSetUserMFAPreference](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_AdminSetUserMFAPreference.html) to either change the MFA preference to a different MFA method or to disable MFA for the user.
+
+</Callout>
+
+In a scenario where MFA is marked as **Required** in Cognito User Pool and another MFA method is not set up, the administrator would need to first initiate an [AdminUpdateUserAttributes](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_AdminUpdateUserAttributes.html) call and update the user’s phone number attribute. Once this is complete, the administrator can continue changing the MFA preference to SMS as suggested above.
+
+## Setting a user's preferred MFA option
+
+<Callout>
+
+The APIs for fetching and updating MFA preferences are only available through AWS Cognito Auth plugin. The examples usage of the API's using the plugin is provided below. 
+
+</Callout>
+
+### Fetch the current user's MFA preferences
+
+Invoke the following API to get the MFA preference for the current user.
+
+import iosFetchMFAPreference from "/src/fragments/lib/auth/ios/mfa_preference/10_fetch_mfa_preference.mdx";
+
+<Fragments fragments={{ios: iosFetchMFAPreference}} />
+
+### Update the current user's MFA preferences
+
+Invoke the following API to update the MFA preference for the current user.
+
+<Callout warning>
+
+Only one MFA method can be marked as preferred at a time. If the user has multiple MFA methods enabled and tries to mark more than one MFA method as preferred, the API will throw an error.
+
+</Callout>
+
+import iosUpdateMFAPreferences from "/src/fragments/lib/auth/ios/mfa_preference/20_update_mfa_preference.mdx";
+
+<Fragments fragments={{ios: iosUpdateMFAPreferences}} />
+
+If multiple MFA methods are enabled for the user, the `signIn` API will return `continueSignInWithMFASelection` as the next step in the auth flow. During this scenario, the user should be prompted to select the MFA method they want to use to sign in.
+
+<Callout>
+
+Refer to the [Continue signin with MFA Selection](/lib/auth/signin_next_steps#continue-signin-with-mfa-selection) section for more details.
```

Comment:
The details for this step should be here - rather than in a seperate page. We should add the code snippet and remove that mention in the other page.

Reply:
the callout would then be removed

Reply:
Well, that page is specifically about enumerating all the steps which can come back from `confirmSignIn`. If we want it on this page, it probably makes sense to have it on both. Otherwise, the other page would be incomplete as a reference.

Reply:
got it - let us have it on both. I am eventually planning on redistributing the content in that page.